src/common/ceph_crypto.h

```diff
@@ -20,61 +21,145 @@
 // with error checking, and just say these really should never fail.
 // This assert MUST NOT be compiled out, even on non-debug builds.
 # include "include/ceph_assert.h"
+#endif /*USE_NSS*/
+
+#ifdef USE_OPENSSL
+typedef struct env_md_ctx_st EVP_MD_CTX;
+typedef struct env_md_st EVP_MD;
```

Comment:
```
In file included from /usr/include/openssl/evp.h:14:0,
                 from /build/ceph-14.0.1-2961-g5770d2c/src/common/ceph_crypto.cc:28:
/usr/include/openssl/ossl_typ.h:91:26: error: conflicting declaration 'typedef struct evp_md_st EVP_MD'
 typedef struct evp_md_st EVP_MD;
                          ^~~~~~

```


```
In file included from /usr/include/openssl/evp.h:14:0,
                 from /build/ceph-14.0.1-2961-g5770d2c/src/common/ceph_crypto.cc:28:
/usr/include/openssl/ossl_typ.h:92:30: error: conflicting declaration 'typedef struct evp_md_ctx_st EVP_MD_CTX'
 typedef struct evp_md_ctx_st EVP_MD_CTX;
                              ^~~~~~~~~~
```

Reply:
It builds correctly for me (and seemingly on CI). Nonetheless I'll push a commit which should clean that up.

Reply:
@jamesba i forgot to post the build failure. see https://jenkins.ceph.com/job/ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=bionic,DIST=bionic,MACHINE_SIZE=huge/17154//consoleFull